### PR TITLE
add email service with nodemailer

### PR DESCRIPTION
**Pull request by eolculnamo2**
- [x] I have read [Chapter's contributing guidelines](https://github.com/freeCodeCamp/chapter/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update README.md`).
- [x] My pull request targets the `master` branch of Chapter.

Creates Email Service using nodemailer that will be used later for our email tickets.

Dependencies Added:
1) nodemailer
2) chai
3) sinon
4) sinon-chai.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -18,7 +18,7 @@
     "speccy:watch": "npx nodemon --watch 'api/swagger.json' --exec 'npm run speccy'",
     "speccy": "npx speccy serve -p 8001 api/swagger.json",
     "start": "npm run prod:server",
-    "test": "NODE_ENV=test npx jest --coverage --verbose",
```
Comment (Haxified) [inline package.json]: it seems you added mocha and chai without documenting it in the pr?

File: server/services/MailerService.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
+import nodemailer, { Transporter, SentMessageInfo } from 'nodemailer';
+import { IMailerService } from 'types/services';
+import Utilities from 'server/util/Utilities';
+
+// @todo add ourEmail, emailUsername, emailPassword, and emailService as
+// environment variables when they become available. Temporary placeholders
+// provided until updated info available.
+export default class MailerService implements IMailerService {
```
Comment (Haxified) [inline server/services/MailerService.ts]: why do you need to implement IMailerService? its not like there are gonna be multiple class definitions following the same guidelines

File: package.json (hunk 1 above)
Comment (eolculnamo2) [reply package.json]: You're absolutely right. I should have mentioned it in the PR. I'm new to contributing to online projects outside of work -- I'll get that updated.

File: server/services/MailerService.ts (hunk 2 above)
Comment (eolculnamo2) [reply server/services/MailerService.ts]: That's true, but the purpose of interfaces is not only to define shared types, but also to show a high level abstraction of a class. In most Object Oriented projects (such as C# and Java), it's generally considered a good practice to include the interface on services and I think it makes sense for NodeJS w/ TS, but I guess its really up to the FCC community. If you all believe I should get rid of it, I certainly will.

File: server/services/MailerService.ts [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    htmlEmail: string,
+    backupText?: string,
+  ) {
+    this.emailList = emailList;
+    this.subject = subject;
+    this.htmlEmail = htmlEmail;
+    this.backupText = backupText || '';
+
+    // to be replaced with env vars
+    this.ourEmail = 'ourEmail@placeholder.place';
```
Comment (vaibhavsingh97) [inline server/services/MailerService.ts]: can you please replace this placeholder with env, and define this in `env.sample`.

Comment (vaibhavsingh97) [reply server/services/MailerService.ts]: Also, please update the readme, to copy env from `env.sample`

Comment (eolculnamo2) [reply server/services/MailerService.ts]: Added :)

File: README.md [hunk 4]
```diff
@@ -80,6 +80,9 @@ Install dependencies:
 npm install
 ```
 
+Setup Environment Variables:
+Copy and paste env.sample into your .env file
```
Comment (ScottBrenner) [inline README.md]: How about using [a ".env" file in Docker Compose](https://docs.docker.com/compose/environment-variables/#the-env-file)?

Comment (eolculnamo2) [reply README.md]: I honestly don't know much at all about Docker -- Do you mean that we should use environment variables in Docker Compose for default variables that can be overwritten by an .env file?

**Comment by allella** [comment]
@vaibhavsingh97 @eolculnamo2 would we need to enable all the transports, by default, to support the common options?

Looks like SMTP is the default and then SES and sendmail (local relay) need to be enabled.

So, would be be installing everything even if people only will use one?

**Comment by jacobbogers** [comment]
Hi, I can look into this

**Comment by jacobbogers** [comment]
Hi fellas, I need to dive into `node-mailer` ( I am only familiar with `mailgun-js`  ), for me to make any sensible decision about design.. I will take this opportunity to learn node-mailer very well,

**Comment by vaibhavsingh97** [comment]
> would we need to enable all the transports, by default, to support the common options?

Nope, by default only SMTP will be enabled, we will just provide options to enable all other providers.

> Would we be installing everything even if people only will use one transport?

Nope, We will give options to users, if they want to use other services.

**Comment by allella** [comment]
> Nope, by default only SMTP will be enabled, we will just provide options to enable all other providers

How would installing new mailer packages work for the app? Would the administrator need to run a command on the server, or is there a way to install new packages directly from the application admin UI and/or configuration files?

**Comment by jacobbogers** [comment]
@allella  nodemailer uses plugins for non SMTP transports (has some build in),  to avoid manually installing them, they need to already exist as a preset (needs its own config page to fill in params) to be selected, I am looking into that as well to put in MVP, it seems to be very trivial.

**Comment by vaibhavsingh97** [comment]
@eolculnamo2 Can you please resolve conflicts so that we can merge 😄

**Comment by eolculnamo2** [comment]
Will do :)

**Comment by vaibhavsingh97** [comment]
Also, Can you please resolve @ScottBrenner comment, also as you had added `mocha` and `chai` for test frameworks, so can you please add the same in the readme [here](https://github.com/freeCodeCamp/chapter#testing) and [here](https://github.com/freeCodeCamp/chapter#tech-stack).

**Comment by allella** [comment]
We had more current conversation in [today's meeting on using nodemailer and SMTP with free / cheap tiers of 3rd party transactional mailing services](https://github.com/freeCodeCamp/chapter/wiki/Meeting-Notes-February-23,-2020#email-service).

**Pull request merged**